### PR TITLE
Add potential evapotranspiration method based on Droogers and Allen (2002)

### LINKED ISSUE #1710: PET calculation with the Hargreaves Modificated method (Droogers and Allen, 2002)
### Generic Issue

* xclim version:  0.39.0
* Python version: 3.8.13
* Operating System: Ubuntu 22.04.02

### Description
In the context of the Copernicus Climate Atlas (https://atlas.climate.copernicus.eu/atlas) we (https://github.com/SantanderMetGroup) are calculating the SPEI with different PET (Potential Evapotranspiration) methods. 

We are using **xclim** during the index computation workflow to ensure process traceability and also because we want to use tools that are aligned with the Copernicus roadmap.

The **xclim.indices.potential_evapotranspiration()** function includes several methods that we are testing, comparing the results with our in-house functions. Currently, we have validated two of the available methods ("thornthwaite48" and "hargreaves85"), and we would like to implement the modificated "hargreaves85" method (Droogers and Allen 2002; https://doi.org/10.1023/A:1015508322413), which includes precipitations as a proxy for humidity (when precipitation occurs, the air contains humidity, thus reducing PET).   

If you are interested in including this new method, we can submit a pull request and provide you with the documentation. Bellow, you'll find our in-house function to calculate the modified method which uses precipitation. This function also implements the "hargreaves85" method because we noticed that the formulation used in xclim is not exactly the same as the one available in the original paper (Hargreaves and Zohrab A. Samani 1985). The xclim formulation (https://github.com/Ouranosinc/xclim/blob/bb39cb1bfde2dbb373180956edf68e5337a5b38d/xclim/indices/_conversion.py#L1285) includes a parameter (lv = 2.5), which differs from the formulation in Hargreaves et al (2002). The inverse of “lv” (1 / 2.5 = 0.4) coincides approximately with the constant 0.408 in Hargreaves et al. (2002)]. This constant is used to convert radiation to evaporation equivalents in mm (kg/MJ).

Many thanks for your help!

```python3
def potential_evapotranspiration(tasmin, tasmax, pr = None, method = 'hg85'):
    """
    Calculate the Potential Evapotranspiration (PET) index using either the HG (Hargreaves 1985) or the MHG (Droogers and Allen 2002) method.

    Parameters
    ----------
    tasmin : xr.Dataset
        Input dataset with daily or monthly 'tasmin'.
    tasmax : xr.Dataset
        Input dataset with daily or monthly 'tasmax'.
    pr : xr.Dataset
        Input dataset with daily or monthly 'pr' for the MHG method. Note that for the MHG method, 
        the suitable temporal scale is monthly. Results might differ when calculated at daily or monthly aggregations 
        (daily-to-monthly aggregated results are NOT identical to monthly ones)
    methods : string
        if 'hg85' then the HG method (Hargreaves 1985) is applied
        if 'mhg' then the MHG method (Droogers and Allen, 2002) is applied
    Returns
    -------
    xarray.Dataset
        Dataset containing the calculated Potential Evapotranspiration index.
    """

    # Extracting temperature variables from the dataset
    tasmin = convert_units_to(tasmin, "degC")
    tasmax = convert_units_to(tasmax, "degC")

    tas = (tasmin + tasmax) / 2        
    tr = tasmax - tasmin
    tr = tr.where(tr>0, 0)
    ra = compute_extraterrestrial_solar_radiation(tasmin, "MJ m-2 d-1")
    
    ra = ra * 0.408 # Is used to convert the radiation to evaporation equivalents in mm (kg/MJ)

    if method == 'mhg':
        # Hargreaves Modificado 2002; Droogers and Allen, 2002 (precipitation in mm/month)
        pr = convert_units_to(pr, "mm/month")
        ab = tr - 0.0123 * pr
        pet_index = (
            0.0013 * ra * (tas + 17.0) * ab ** 0.76
        )
        pet_index = xr.where(np.isnan(ab**0.76), 0, pet_index)

    elif method == 'hg85':
        # Hargreaves et al. (1985) and Hargreaves (1994)
        pet_index = 0.0023 * ra * (tas + 17.8) * tr ** 0.5

    pet_index = pet_index.clip(0)
    pet_index = pet_index.to_dataset(name="pet")
    pet_index['pet'].attrs['units'] = 'mm'
        
    return pet_index
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by JavierDiezSierra**
### Pull Request Checklist:
- [X] This PR addresses an already opened issue (for bug fixes / features)
  - This PR closes #1710 
- [X] Tests for the changes have been added (for bug fixes / features)
  - [X] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [X] CHANGES.rst has been updated (with summary of main changes)
  - [X] Link to issue (:issue:`1710`) and pull request (:pull:`1723`) has been added

### What kind of change does this PR introduce?

This change introduces a new method to calculate potential evapotranspiration based on Droogers and Allen (2002). The method incorporates precipitation as a proxy for humidity and is more appropriate for dry regions

### Does this PR introduce a breaking change?

No

### CODE REVIEW

**Review by aulemahal — COMMENTED** [review]
Thanks for the PR!

Please look at my commit to see if you agree. We try to avoid exact frequency matching in indices. Xclim's Indicators are the objects that may enforce strong rules about data frequency, but indices should let the user do what it wants.

So instead of deciding what to do according to the `infer_freq`, I used the same idea as in the TW48 implementation : resample to monthly no matter what. If the data is already monthly, this will be a no-op. I also generalized the function to get a daily time coord from monthly data, as used by TW48 also. 

One caveat of my technique is the chunksizes that are not easily inferable. I took a guess that passing the same chunks as monthly tasmin would be ok in most cases. Especially because we resample the radiation right away.

Would you be able to add a test case for this new method ? The test should be a method of  `class TestPotentialEvapotranspiration`,  line 3159 of `tests/test_indices.py`.

File: CHANGES.rst [hunk 1]
```diff
@@ -4,12 +4,16 @@ Changelog
...
-Contributors to this version: Trevor James Smith (:user:`Zeitsperre`), Pascal Bourgault (:user:`aulemahal`), Juliette Lavoie (:user:`juliettelavoie`), David Huard (:user:`huard`), Gabriel Rondeau-Genesse (:user:`RondeauG`).
+Contributors to this version: Trevor James Smith (:user:`Zeitsperre`), Pascal Bourgault (:user:`aulemahal`), Juliette Lavoie (:user:`juliettelavoie`), David Huard (:user:`huard`), Gabriel Rondeau-Genesse (:user:`RondeauG`), Javier Diez-Sierra (:user:`JavierDiezSierra`).
 
 Announcements
 ^^^^^^^^^^^^^
 * `xclim` has migrated its development branch name from `master` to `main`. (:issue:`1667`, :pull:`1669`).
 
+New features and enhancements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+* Indicator ``xclim.indices.potential_evapotranspiration`` now accepts a new value (`DA02`) for argument `method` implementing potential evapotranspiration based on Droogers and Allen (2002). (:issue:`1710`, :pull:`1723`).
```
[the remark this answers is not in the document]
Comment (Zeitsperre) [reply CHANGES.rst:15]:
```suggestion
* Indicator ``xclim.atmos.potential_evapotranspiration`` and indice ``xclim.indices.potential_evapotranspiration`` now accept a new value (`DA02`) for argument `method` implementing potential evapotranspiration based on Droogers and Allen (2002). (:issue:`1710`, :pull:`1723`).
```

**Comment by Zeitsperre** [comment]
@aulemahal I think that I have the workflows configured to trigger on any comment. This might be why the builds often cancel during reviews. Will open an issue.

File: docs/references.bib [hunk 2]
```diff
@@ -2114,3 +2114,19 @@ @article{brode_utci_2012
...
+	journal = {Irrigation and Drainage Systems},
+	volume = {16},
+	number = {1},
+	pages = {33 – 45},
+	doi = {10.1023/A:1015508322413},
+	url = {https://www.scopus.com/inward/record.uri?eid=2-s2.0-0036464359&doi=10.1023%2fA%3a1015508322413&partnerID=40&md5=7322aaa4c6874878f5b1dab3c73c1718},
+	type = {Article},
+	publication_stage = {Final},
+	source = {Scopus},
+	note = {Cited by: 747}
```
Comment (Zeitsperre) [inline docs/references.bib:2131]:
```suggestion
	type = {Article}
```

**Review by Zeitsperre — APPROVED** [review]

**Comment by JavierDiezSierra** [comment]
> Thanks for the PR!
> 
> Please look at my commit to see if you agree. We try to avoid exact frequency matching in indices. Xclim's Indicators are the objects that may enforce strong rules about data frequency, but indices should let the user do what it wants.
> 
> So instead of deciding what to do according to the `infer_freq`, I used the same idea as in the TW48 implementation : resample to monthly no matter what. If the data is already monthly, this will be a no-op. I also generalized the function to get a daily time coord from monthly data, as used by TW48 also.
> 
> One caveat of my technique is the chunksizes that are not easily inferable. I took a guess that passing the same chunks as monthly tasmin would be ok in most cases. Especially because we resample the radiation right away.
> 
> Would you be able to add a test case for this new method ? The test should be a method of `class TestPotentialEvapotranspiration`, line 3159 of `tests/test_indices.py`.

Thank you for the proposed changes. It seems correct to resample everything to monthly and work at that resolution. Regarding the issue of chunksize when calculating solar radiation, it was causing problems for me, so I have removed it. On the other hand, I have modified the term lv=2.5 and included the constant 0.408 to align the Hargreaves (HG85) method with the new modified method (DA02). The result of applying 1/2.5 = 0.4 is very similar to multiplying by 0.408, but this way we use the original constant that appears in Hargreaves et al (2002).

Finally, I have added the test for the new method, although when running all tests with `pytest`, I encountered many errors in many other tests, although perhaps it's my problem and I haven't followed the correct procedure.

**Review by Zeitsperre — APPROVED** [review]

**Comment by Zeitsperre** [comment]
@JavierDiezSierra This is looking good!

Once the failing tests are addressed, we should be able to move forward with the merge in advance of the next release (v0.49.0).

All the best!

**Comment by aulemahal** [comment]
@JavierDiezSierra I think I fixed the tests in the last commit. This looks good to go.

**Comment by JavierDiezSierra** [comment]
> @JavierDiezSierra I think I fixed the tests in the last commit. This looks good to go.

¡Many thanks for you help!.

**Comment by Zeitsperre** [comment]
@JavierDiezSierra

Thanks so much for your pull request!

**Pull request merged**